src/ApplePayGateway.php

```diff
@@ -0,0 +1,206 @@
+ *    // response body will be empty.
+ *    if ($authorizeResponse->isSuccessful()) {
+ *        echo 'Status Code: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *        echo 'Status Reason: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *        echo 'Apple Pay Payment Session Object: ' . $authorizeResponse->getPaymentSessionObject()
+ *          . PHP_EOL;
+ *        echo 'Response: ' . $authorizeResponse->getResponse() . PHP_EOL;
+ *    } else {
+ *        // Error handling.
+ *        echo 'ERROR: Request not successful.';
+ *        echo 'Status Code: ' . $authorizeResponse->getCode() . PHP_EOL;
+ *        echo 'Status Message: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *    }
+ *
+ *    //Pass authorizeResponse back to the client to validate your merchant and continue with an Apple Pay payment.
+ *    //If successful, the payment sheet should be fully loaded.
+ *    $apple_pay_session = $authorizeResponse->getPaymentSessionObject();
+ *
+ *     // TODO: Add functionality for completeAuthorize() and capture().
+ *     // An opaque Apple Pay Session is returned as a response (expires after 5 mins) and it can be sent to
+ *     // the front end to fully load the payment sheet. This allows the user to optionally configure their
+ *     // payment option and shipping methods (if needed) and submit their payment.
+ * </code>
+ */
+class ApplePayGateway extends AbstractVindiciaGateway
+{
+
+    /**
+     * Get the gateway name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return 'Vindicia ApplePay';
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getPemCertPath()
+    {
+        return $this->getParameter('pemCertPath');
+    }
+
+    /**
+     * A certificate associated with your merchant ID.
+     * Example:
+     * setPemCertPath('./certs/path_to_cert.crt.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setPemCertPath($value)
+    {
+        return $this->setParameter('pemCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPath()
+    {
+        return $this->getParameter('keyCertPath');
+    }
+
+    /**
+     * A key certificate associated with your merchant ID.
+     * Example:
+     * setKeyCertPath('./certs/path_to_cert.key.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPath($value)
+    {
+        return $this->setParameter('keyCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPassword()
+    {
+        return $this->getParameter('keyCertPassword');
+    }
+
+    /**
+     * Password associated with your key certificate.
+     * Can be omitted if a password isn't set.
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPassword($value)
```

Comment:
Could you add typehints to all function params and return values. Some functions have them and others don't so better to be consistent.

Reply:
I'm not sure that will work because the value of these setters depends on the context of the person using them.

Reply:
do you mean php 7 style typehints? or @param annotations? omnipay supports php 5, so we can't use the php 7 style typehints. (`array` is supported in php 5.) if there are any missing @param annotations, those should definitely be addressed. just let danaya know where you saw them

Reply:
Ahh, my bad. I thought type hints were already part of php 5